### ISSUE TITLE
Import fails on poorly formatted tracks

### ISSUE THREAD

**Issue by 1xPdd** [spl0k/supysonic#132]
Hi, I'm trying to import my library to Supysonic but it ends up failing on the initial scan.

```
Scanning 'Music': 14166 files scannedTraceback (most recent call last):
  File "/usr/bin/supysonic-cli", line 23, in <module>
    cli.onecmd(' '.join(sys.argv[1:]))
  File "<string>", line 2, in onecmd
  File "/usr/lib/python3.6/site-packages/pony/orm/core.py", line 460, in new_func
    try: return func(*args, **kwargs)
  File "/usr/lib/python3.6/cmd.py", line 217, in onecmd
    return func(arg)
  File "/usr/lib/python3.6/site-packages/supysonic/cli.py", line 67, in method
    return func(** { key: vars(args)[key] for key in vars(args) if key != 'action' })
  File "/usr/lib/python3.6/site-packages/supysonic/cli.py", line 175, in folder_scan
    scanner.scan(folder, TimedProgressDisplay(folder.name, self.stdout))
  File "/usr/lib/python3.6/site-packages/supysonic/scanner.py", line 72, in scan
    self.scan_file(full_path)
  File "<string>", line 2, in scan_file
  File "/usr/lib/python3.6/site-packages/pony/orm/core.py", line 460, in new_func
    try: return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/supysonic/scanner.py", line 154, in scan_file
    Track(**trdict)
  File "/usr/lib/python3.6/site-packages/supysonic/db.py", line 46, in __init__
    db.Entity.__init__(self, *args, **kwargs)
  File "<string>", line 2, in __init__
  File "/usr/lib/python3.6/site-packages/pony/utils/utils.py", line 58, in cut_traceback
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pony/orm/core.py", line 4320, in __init__
    avdict[attr] = attr.validate(val, obj, entity, from_db=False)
  File "/usr/lib/python3.6/site-packages/pony/orm/core.py", line 2356, in validate
    attr if obj is None or obj._status_ is None else '%r.%s' % (obj, attr.name)))
  File "/usr/lib/python3.6/site-packages/pony/utils/utils.py", line 98, in throw
    raise exc
ValueError: Attribute Track.title is required
```

I'm guessing the title track is missing or corrupted. Where the import fails seems to vary, so the 14166 is not constant. Additionally, in beets, Cantata (mpd), and Quod Libet, I can't find a track without a title, making further investigation difficult.

It would be ideal of such failures were either skipped or imported with the offending tag simply blank.

Thanks.

**Comment by spl0k** [comment]
This kind of error was fixed some time ago (918cd11262a15c581d9ea88bb399955e912fe940). Empty or missing track titles are replaced by the file name.
Are you sure your installation is up-to-date with the current _master_? The line numbers in your error don't match the current version of the code.

**Comment by 1xPdd** [comment]
Ahh, my bad. You're right. I was using a docker container that, it seems, was not very up-to-date. Changing to a newer one solved my problem. I'm now enjoying my music liberated from java. Thanks!
